lib/rafka/consumer.rb

```diff
@@ -185,7 +192,15 @@ def commit(*msgs)
       tp.each do |topic, po|
         po.each do |partition, offset|
           Rafka.wrap_errors do
-            @redis.rpush("acks", "#{topic}:#{partition}:#{offset}")
+            begin
+              @redis.rpush("acks", "#{topic}:#{partition}:#{offset}")
+            rescue Redis::CommandError => e
+              if e.message.start_with?("CONS No consumer registered for Client")
+                @logger.warn("CONS Cannot commit offsets, connection with rafka was temporarly lost.")
+              else
+                raise e
```

Comment:
Since this is a library I wouldn't go as far as setting a logger here.

A more un-opinionated approach, imho, is to create the following error in `lib/error.rb` and re-raise it letting the library user handle it and possibly log whatever they want:

```
class UregisteredConsumerError < ConsumeError; end
```

```
rescue Redis::CommandError => e
  if e.message.start_with?("CONS No consumer registered for Client")
     raise UregisteredConsumerError.new(e.message)
  else
    raise e
  end
```

what do you think?

Reply:
We could also utilize the `wrap_errors` function in order not to redefined `Redis::CommandError` error handlings again

```ruby
po.each do |partition, offset|
  Rafka.wrap_errors do
    @redis.rpush("acks", "#{topic}:#{partition}:#{offset}")
  end
rescue ConsumeError => e
  raise e unless e.message.start_with?("No consumer registered for Client")

  raise UnregisteredConsumerError.new(e.message)
end
```

Reply:
My intention was to address this at one place for all the requests.  If we don't want that, I don't see any value of reraising the error with a different type here. The consumer will have to catch it and handle it in every consume call. In this is the case, I would leave the library as is and filter the original error in client's side every time it calls consume, or commit interfaces.